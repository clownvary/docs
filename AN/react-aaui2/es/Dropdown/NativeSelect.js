import _extends from 'babel-runtime/helpers/extends';
import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import React, { Component } from 'react';
import { string, bool, object, func, oneOf, arrayOf, shape } from 'prop-types';
import classNames from 'classnames';
import { filterValidProps, noop, omit } from '../shared/utils';
import * as da from '../shared/data-access';
import injectL10n from '../shared/injectL10n';
import { aauiL10nShape } from '../shared/types';

var Select = function (_Component) {
  _inherits(Select, _Component);

  function Select() {
    var _temp, _this, _ret;

    _classCallCheck(this, Select);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, _Component.call.apply(_Component, [this].concat(args))), _this), _this.handleSelectChange = function (e) {
      var value = e.target.value;
      var onChange = _this.props.onChange;


      onChange({ value: value });
    }, _this.render = function () {
      var _this$props = _this.props,
          className = _this$props.className,
          disabled = _this$props.disabled,
          preIcon = _this$props.preIcon,
          placeholder = _this$props.placeholder,
          size = _this$props.size,
          theme = _this$props.theme,
          isMoreButton = _this$props.isMoreButton,
          style = _this$props.style,
          data = _this$props.data,
          l10n = _this$props.l10n,
          rest = _objectWithoutProperties(_this$props, ['className', 'disabled', 'preIcon', 'placeholder', 'size', 'theme', 'isMoreButton', 'style', 'data', 'l10n']);

      var validProps = filterValidProps(rest);

      var classes = classNames('dropdown', 'dropdown--native', 'dropdown--' + theme, 'dropdown--' + size, {
        'dropdown--disabled': disabled
      }, className);
      var selectClasses = classNames({
        dropdown__button: true,
        dropdown__select: true,
        disabled: disabled
      });

      return React.createElement(
        'div',
        _extends({ className: classes, style: style }, validProps),
        preIcon && React.createElement('i', { className: 'dropdown__prefix-icon ' + preIcon }),
        React.createElement(
          'select',
          _extends({
            className: selectClasses,
            onChange: _this.handleSelectChange,
            disabled: disabled
          }, omit(rest, ['onChange', 'uncontrolled', 'filterPlaceholder', 'highlight', 'maxHeight'])),
          !rest.value && !rest.defaultValue && React.createElement(
            'option',
            { value: 'prompt' },
            placeholder || l10n.formatMessage('react-aaui.common.dropdown.notice.select')
          ),
          data.map(function (item) {
            return React.createElement(
              'option',
              { value: item.value, key: item.value },
              da.get(item, 'text')
            );
          })
        ),
        isMoreButton ? React.createElement('span', { className: 'icon-caret-down' }) : React.createElement('span', { className: 'icon-chevron-down' })
      );
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  return Select;
}(Component);

Select.propTypes = {
  className: string,
  value: string,
  defaultValue: string,
  placeholder: string,
  preIcon: string,
  size: oneOf(['m', 'lg']),
  theme: oneOf(['flat', 'gradient', 'borderless']),
  isMoreButton: bool,
  disabled: bool,
  data: arrayOf(shape({
    text: string,
    value: string
  })),
  style: object, // eslint-disable-line
  onChange: func,
  l10n: aauiL10nShape
};
Select.defaultProps = {
  size: 'm',
  theme: 'flat',
  onChange: noop
};


export default injectL10n()(Select);