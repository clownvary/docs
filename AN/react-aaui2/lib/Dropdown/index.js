'use strict';

exports.__esModule = true;
exports.Select = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Dropdown = require('./Dropdown');

var _Dropdown2 = _interopRequireDefault(_Dropdown);

var _NativeSelect = require('./NativeSelect');

var _NativeSelect2 = _interopRequireDefault(_NativeSelect);

var _viewport = require('../viewport');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Select = function Select(props) {
  return _react2.default.createElement(
    'div',
    null,
    _react2.default.createElement(
      _viewport.ShowAt,
      { smAndAbove: true },
      _react2.default.createElement(_Dropdown2.default, props)
    ),
    _react2.default.createElement(
      _viewport.HideAt,
      { smAndAbove: true },
      _react2.default.createElement(_NativeSelect2.default, props)
    )
  );
};

exports.Select = Select;
exports.default = _Dropdown2.default;