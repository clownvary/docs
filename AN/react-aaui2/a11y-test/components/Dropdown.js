import React from 'react'
import ReactDOM from 'react-dom'
import { Dropdown } from '../../src/index'

class App extends React.Component {
  state = {
    val: 'chs',
    d1: [
      { text: 'Canada', value: 'can' },
      { text: 'China', value: 'chs' },
      { text: 'Japan', value: 'jap' },
      { text: 'United States of America', value: 'usa' },
    ],
    isChecked: false,
  }

  showValue = ({ value }) => {
    console.log(`You selected: ${value}`)
  }

  showValue2 = () => {
    console.log(this.d1.value)
  }

  render() {
    const data1 = [
      { text: 'Canada', value: 'can' },
      { text: 'China', value: 'chs' },
      { text: 'Japan1', value: 'jap1' },
      { text: 'Japan2', value: 'jap2' },
      { text: 'Japan3', value: 'jap3' },
      { text: 'Japan4', value: 'jap4' },
      { text: 'Japan5', value: 'jap5' },
      { text: 'Japan6', value: 'jap6' },
      { text: 'Japan7', value: 'jap7' },
      { text: 'Japan8', value: 'jap8' },
      { text: 'Japan9', value: 'jap9' },
      { text: 'Japan10', value: 'jap10' },
      { text: 'Japan11', value: 'jap11' },
      { text: 'Japan12', value: 'jap12' },
      { text: 'Japan13', value: 'jap13' },
      { text: 'Japan14', value: 'jap14' },
      { text: 'Japan15', value: 'jap15' },
      { text: 'Japan16', value: 'jap16' },
      { text: 'Japan17', value: 'jap17' },
      { text: 'Japan18', value: 'jap18' },
      { text: 'Japan19', value: 'jap19' },
      { text: 'Japan20', value: 'jap20' },
      { text: 'Japan21', value: 'jap21' },
      { text: 'Japan22', value: 'jap22' },
      { text: 'Japan23', value: 'jap23' },
      { text: 'Japan24', value: 'jap24' },
      { text: 'Japan25', value: 'jap25' },
      { text: 'Japan26', value: 'jap26' },
      { text: 'Japan27', value: 'jap27' },
      { text: 'United States of America', value: 'usa' },
    ]

    const data2 = new Array(100)
    for (let i = 1; i <= 100; i += 1) {
      data2[i - 1] = { text: i.toString(), value: i }
    }

    return (
      <div style={{ fontFamily: 'ProximaNova, Arial, sans-serif', color: 'gray' }}>
        <Dropdown name="d1" ref={input => { this.d1 = input }} data={this.state.d1} placeholder="Choose..." onChange={this.showValue2} />
        <br />
        <Dropdown data={data1} defaultValue="usa" onChange={this.showValue} />
        <br />
      </div>
    )
  }
}

ReactDOM.render(<App />, document.getElementById('root'))
