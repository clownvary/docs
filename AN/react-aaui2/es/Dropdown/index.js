import React from 'react';
import Dropdown from './Dropdown';
import NativeSelect from './NativeSelect';
import { ShowAt, HideAt } from '../viewport';

var Select = function Select(props) {
  return React.createElement(
    'div',
    null,
    React.createElement(
      ShowAt,
      { smAndAbove: true },
      React.createElement(Dropdown, props)
    ),
    React.createElement(
      HideAt,
      { smAndAbove: true },
      React.createElement(NativeSelect, props)
    )
  );
};

export { Select };
export default Dropdown;