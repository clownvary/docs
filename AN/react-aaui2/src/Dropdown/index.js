import React from 'react'
import Dropdown from './Dropdown'
import NativeSelect from './NativeSelect'
import { ShowAt, HideAt } from '../viewport'

const Select = props => (
  <div>
    <ShowAt smAndAbove>
      <Dropdown {...props} />
    </ShowAt>
    <HideAt smAndAbove>
      <NativeSelect {...props} />
    </HideAt>
  </div>
)

export { Select }
export default Dropdown
