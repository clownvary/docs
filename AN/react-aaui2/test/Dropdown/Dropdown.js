import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import Dropdown from '../../src/Dropdown/Dropdown'

const simpleData = [
  {
    text: 'China',
    value: 'chs',
  },
  {
    text: 'United States of America',
    value: 'usa',
  },
]

const App = () => {
  const data = [
    {
      text: 'Canada',
      value: 'can',
    },
    {
      text: 'China',
      value: 'chs',
    },
    {
      text: 'Japan',
      value: 'jap',
    },
    {
      text: 'United States of America',
      value: 'usa',
    },
  ]

  return <Dropdown data={data} />
}

const AppWithFilter = () => {
  const data = [
    { text: 'Canada', value: 'can' },
    { text: 'China', value: 'chs' },
    { text: 'Japan', value: 'jap' },
    { text: 'United States of America', value: 'usa' },
  ]

  return <Dropdown data={data} filter className="dropdown--with-search" />
}

it("App should render right thing and don't change unexpected", () => {
  const tree = renderer.create(<App />).toJSON()

  expect(tree).toMatchSnapshot()
})

it("AppWithFilter should render right thing and don't change unexpected", () => {
  const tree = renderer.create(<AppWithFilter />).toJSON()

  expect(tree).toMatchSnapshot()
})

it("A dropdown's menu should hide by default", () => {
  const wrapper = mount(<App />)
  expect(wrapper.find('.dropdown__button').hasClass('collapse')).toBe(true)
})

it("A dropdown's menu should show up when click button", () => {
  const wrapper = mount(<App />)
  wrapper.find('.dropdown__button').simulate('click')
  expect(wrapper.find('.dropdown__button').hasClass('expand')).toBe(true)
})

it("A dropdown's menu should hide again when click button twice", () => {
  const wrapper = mount(<App />)
  wrapper.find('.dropdown__button').simulate('click')
  wrapper.find('.dropdown__button').simulate('click')
  expect(wrapper.find('.dropdown__button').hasClass('collapse')).toBe(true)
})

it("A dropdown's menu should hide when dropdown menu lost focus", () => {
  const wrapper = mount(<App />)
  wrapper.find('.dropdown__menu').simulate('blur')
  expect(wrapper.find('.dropdown__button').hasClass('collapse')).toBe(true)
})

it('Should show prefix icon if preIcon prop has a value', () => {
  const wrapper = mount(<Dropdown data={simpleData} />)

  wrapper.setProps({
    preIcon: 'icon-location',
  })

  expect(wrapper.find('.dropdown__prefix-icon').length).toBe(1)
})

it('Default option should be highlighted color with a `highlight` prop', () => {
  const wrapper = mount(
    <Dropdown data={simpleData} highlight defaultValue="chs" />,
  )
  expect(
    wrapper
      .find('li')
      .at(0)
      .hasClass('selected'),
  ).toBe(true)
})

it('Default option should not be highlighted color without a `highlight` prop', () => {
  const wrapper = mount(<Dropdown data={simpleData} defaultValue="chs" />)
  expect(
    wrapper
      .find('li')
      .at(0)
      .hasClass('selected'),
  ).toBe(false)
})

it('Highlighted color should update when defaultValue changes', () => {
  const wrapper = mount(<Dropdown data={simpleData} highlight />)

  wrapper.setProps({
    defaultValue: 'usa',
  })

  expect(
    wrapper
      .find('li')
      .at(0)
      .hasClass('selected'),
  ).toBe(false)
  expect(
    wrapper
      .find('li')
      .at(1)
      .hasClass('selected'),
  ).toBe(true)
})

it('Highlighted color should update when value changes', () => {
  const wrapper = mount(
    <Dropdown data={simpleData} highlight defaultValue="chs" />,
  )

  wrapper.setProps({
    value: 'usa',
  })

  expect(
    wrapper
      .find('li')
      .at(0)
      .hasClass('selected'),
  ).toBe(false)
  expect(
    wrapper
      .find('li')
      .at(1)
      .hasClass('selected'),
  ).toBe(true)
})

it('Highlighted color should update when data changes', () => {
  const wrapper = mount(<Dropdown data={[]} highlight defaultValue="chs" />)

  wrapper.setProps({
    data: simpleData,
  })

  expect(
    wrapper
      .find('li')
      .at(0)
      .hasClass('selected'),
  ).toBe(true)
})

it('Current option should be highlighted color with a `highlight` prop', () => {
  const wrapper = mount(<Dropdown data={simpleData} highlight />)
  const selectedLi = wrapper.find('li').at(1)

  selectedLi.find('a').simulate('click')

  expect(
    wrapper
      .find('li')
      .at(1)
      .hasClass('selected'),
  ).toBe(true) // https://github.com/airbnb/enzyme/issues/1163
})

it('Current option should not be highlighted color without a `highlight` prop', () => {
  const wrapper = mount(<Dropdown data={simpleData} />)
  const selectedLi = wrapper.find('li').at(1)

  selectedLi.find('a').simulate('click')
  wrapper.update()

  expect(selectedLi.hasClass('selected')).toBe(false)
})

it('A dropdown text should be set after select menu', () => {
  const wrapper = mount(<App />)
  wrapper.find('.dropdown__button').simulate('click')
  wrapper.find('ul').simulate('keyDown', { keyCode: 40 })
  wrapper.find('ul').simulate('keyDown', { keyCode: 38 })
  wrapper.find('ul').simulate('keyDown', { keyCode: 13 })
  wrapper
    .find('ul')
    .childAt(1)
    .find('a')
    .simulate('click')
  expect(wrapper.find('.dropdown__button-text').text()).toBe(
    wrapper
      .find('ul')
      .childAt(1)
      .find('a')
      .text(),
  )
})

it('A dropdown with filter should can search', () => {
  const wrapper = mount(<AppWithFilter />)
  wrapper.find('button').simulate('click')
  wrapper.find('.input').simulate('change', { target: { value: 'Cana' } })
  wrapper
    .find('ul')
    .childAt(2)
    .find('a')
    .simulate('click')
  expect(wrapper.find('.dropdown__button-text').text()).toBe(
    wrapper
      .find('ul')
      .childAt(2)
      .find('a')
      .text(),
  )
})

it('Should show dropdown menu after pressed the blank key (keyCode=32)', () => {
  const wrapper = mount(<App />)
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 32 })
  expect(wrapper.find('ul').hasClass('show')).toBe(true)
})

it('Should show dropdown menu after pressed the Up key (keyCode=38)', () => {
  const wrapper = mount(<App />)
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 38 })
  expect(wrapper.find('ul').hasClass('show')).toBe(true)
})

it('Should show dropdown menu after pressed the Down key (keyCode=40)', () => {
  const wrapper = mount(<App />)
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 40 })
  expect(wrapper.find('ul').hasClass('show')).toBe(true)
})

it('Should hide dropdown menu after pressed the Esc key (keyCode=27)', () => {
  const wrapper = mount(<App />)
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 32 })
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 27 })
  expect(wrapper.find('ul').hasClass('show')).toBe(false)

  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 38 })
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 27 })
  expect(wrapper.find('ul').hasClass('show')).toBe(false)

  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 40 })
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 27 })
  expect(wrapper.find('ul').hasClass('show')).toBe(false)
})

it('Should hide dropdown menu after pressed the Enter key (keyCode=13)', () => {
  const wrapper = mount(<App />)

  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 38 })
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 13 })
  expect(wrapper.find('ul').hasClass('show')).toBe(false)

  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 40 })
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 13 })
  expect(wrapper.find('ul').hasClass('show')).toBe(false)

  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 32 })
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 38 })
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 13 })
  expect(wrapper.find('ul').hasClass('show')).toBe(false)
})

it('Should active the first item after pressed the Up key (keyCode=38)', () => {
  const wrapper = mount(<App />)
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 38 })
  expect(
    wrapper
      .find('li')
      .last()
      .hasClass('selected'),
  ).toBe(true)
})

it('Should active the last item after pressed the Down key (keyCode=40)', () => {
  const wrapper = mount(<App />)
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 40 })
  expect(
    wrapper
      .find('li')
      .first()
      .hasClass('selected'),
  ).toBe(true)
})

it('Should active the second item after pressed the Down key (keyCode=40) and press Down key again', () => {
  const wrapper = mount(<App />)
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 40 })
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 40 })
  expect(
    wrapper
      .find('li')
      .at(1)
      .hasClass('selected'),
  ).toBe(true)
})

it('The text of button eaquals the selectd li text after pressed the Down key (keyCode=40) and press Down key again and press the Enter key ', () => {
  const wrapper = mount(<App />)
  wrapper.find('.dropdown__button').simulate('click')
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 40 })
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 40 })
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 13 })
  expect(
    wrapper
      .find('.dropdown__button')
      .find('span')
      .first()
      .text(),
  ).toBe('China')
})

it('Should active the second item after pressed the Down key (keyCode=40) and press J key, then press U key', () => {
  const wrapper = mount(<App />)
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 40 })
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 74 }) // Start with J
  expect(
    wrapper
      .find('li')
      .at(2)
      .hasClass('selected'),
  ).toBe(true)
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 85 }) // Start with U
  expect(
    wrapper
      .find('li')
      .at(3)
      .hasClass('selected'),
  ).toBe(false)
  setTimeout(() => {
    wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 85 }) // Start with U
    expect(
      wrapper
        .find('li')
        .at(3)
        .hasClass('selected'),
    ).toBe(true)
  }, 500)
})

it('Should active the second item after pressed the Down key (keyCode=40) and press E key', () => {
  const wrapper = mount(<App />)
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 40 })
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 69 }) // Start with E
  expect(
    wrapper
      .find('li')
      .at(0)
      .hasClass('selected'),
  ).toBe(true) // Still selected the first item
})

it("drop down's value can be updated from code directly ", () => {
  const wrapper = mount(<Dropdown data={simpleData} />)
  wrapper.find('.dropdown__button').simulate('click')
  wrapper
    .find('ul')
    .childAt(1)
    .find('a')
    .simulate('click')
  const originalResult = wrapper.instance().value
  wrapper.setProps({
    value: undefined,
  })
  wrapper.instance().value = 'chs'
  expect(wrapper.instance().value).not.toBe(originalResult)
})

it("automatically improve drop down's scrollTop when rendering ", () => {
  const wrapper = mount(<Dropdown data={simpleData} />)
  wrapper.instance().itemMenu.querySelector = () => ({
    offsetTop: 20,
    offsetHeight: 10,
    focus: () => {},
  })
  wrapper.setState({
    isExpanded: true,
  })
  wrapper.setProps({
    placeholder: 'ok',
  })
  expect(wrapper.instance().itemMenu.scrollTop).toBe(20)
})

it('can collapse and cancel it', () => {
  jest.useFakeTimers()
  const wrapper = mount(<Dropdown data={simpleData} />)
  wrapper.setState({
    isExpanded: true,
  })
  wrapper.instance().giveFocus({
    target: wrapper.instance().itemMenu,
  })
  wrapper.instance().tryCollapse()
  jest.runAllTimers()
  wrapper.instance().cancelCollapseTimeout()
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 32 })
  wrapper.find('.dropdown__button').simulate('keyDown', { keyCode: 9 })
  wrapper.find('.dropdown__button').simulate('mouseDown')
  expect(wrapper.find('ul').hasClass('show')).toBe(true)
})

it('can handle keydown when filtered', () => {
  const wrapper = mount(<Dropdown data={simpleData} />)
  wrapper.setProps({
    filter: true,
  })
  wrapper
    .find('.dropdown__menu__search-box input')
    .simulate('keyDown', { keyCode: 38 })
  wrapper
    .find('.dropdown__menu__search-box input')
    .simulate('keyDown', { keyCode: 13 })
  wrapper
    .find('.dropdown__menu__search-box input')
    .simulate('keyDown', { keyCode: 10 })
  expect(wrapper.find('ul').hasClass('show')).toBe(false)
})

it('can handle keydown when filtered case 2', () => {
  const wrapper = mount(<Dropdown data={simpleData} />)
  wrapper.setProps({
    filter: true,
  })
  wrapper.instance().filterData('', simpleData)
  expect(wrapper.find('ul').hasClass('show')).toBe(false)
})

it('should clear keyboard value when trigger keyUp on the wrapper', () => {
  const wrapper = mount(<Dropdown data={simpleData} />)
  wrapper.setProps({
    disabled: false,
  })
  wrapper
    .find('div.dropdown')
    .simulate('keyUp', { keyCode: 38, persist: () => true })
  expect(wrapper.instance().keyboardValue).toBe('')
})

it('should handle keyUp event from the wrapper correctly', () => {
  const wrapper = mount(<Dropdown data={simpleData} />)
  const keyboardValue = 57
  wrapper.setProps({
    disabled: false,
    data: [
      { text: '5', value: 5 },
      { text: '7', value: 7 },
      { text: '57', value: 57 },
    ],
  })
  wrapper.setState({
    keyWords: '5',
    activeItemIndex: 4,
  })
  wrapper.instance().keyboardValue = '5'
  wrapper.find('div.dropdown').simulate('keyDown', { keyCode: 55 })
  expect(wrapper.instance().keyboardValue).toBe(keyboardValue.toString())
})

it('should handle keyUp event from the wrapper correctly 2', () => {
  const wrapper = mount(<Dropdown data={simpleData} />)
  const keyboardValue = 5
  wrapper.setProps({
    disabled: false,
    data: [
      { text: '5', value: 5 },
      { text: '7', value: 7 },
      { text: '57', value: 57 },
    ],
  })
  wrapper.setState({
    keyWords: '5',
    activeItemIndex: 4,
  })
  wrapper.instance().keyboardValue = '5'
  wrapper.find('div.dropdown').simulate('keyDown', { keyCode: 38 })
  expect(wrapper.instance().keyboardValue).toBe(keyboardValue.toString())
})

it('should not handle keyUp event when disbaled', () => {
  const wrapper = mount(<Dropdown data={simpleData} />)
  wrapper.setProps({
    disabled: true,
  })
  wrapper.find('div.dropdown').simulate('keyDown', { keyCode: 55 })
  expect(wrapper.instance().keyboardValue).toBe('')
})

it('should show initValue correctly', () => {
  const wrapper = mount(<Dropdown data={simpleData} />)
  wrapper.setProps({
    disabled: true,
    defaultValue: 'usa',
  })
  wrapper.find('div.dropdown').simulate('keyDown', { keyCode: 55 })
  expect(wrapper.instance().keyboardValue).toBe('')
})

it('should show initValue correctly without state change case 1', () => {
  const wrapper = mount(<Dropdown data={simpleData} />)
  wrapper.setState({
    value: '5',
  })
  wrapper.setProps({
    disabled: true,
    defaultValue: 'usa',
  })
  wrapper.find('div.dropdown').simulate('keyDown', { keyCode: 38 })
  expect(wrapper.instance().keyboardValue).toBe('')
})

it('should show initValue correctly without state change case 2', () => {
  const wrapper = mount(<Dropdown data={simpleData} />)
  wrapper.setState({
    isExpanded: true,
  })
  wrapper.find('div.dropdown').simulate('keyDown', { keyCode: 32 })
  expect(wrapper.instance().keyboardValue).toBe('')
})

it('should call onExpand when state set isExpanded to true', () => {
  const onExpand = jest.fn()
  const wrapper = mount(<Dropdown data={simpleData} onExpand={onExpand} />)
  wrapper.setState({
    isExpanded: true,
  })
  expect(onExpand).toBeCalled()
})

it('should call onCollapse when state set isExpanded to false from true', () => {
  const onExpand = jest.fn()
  const onCollapse = jest.fn()
  const wrapper = mount(
    <Dropdown data={simpleData} onExpand={onExpand} onCollapse={onCollapse} />,
  )
  wrapper.setState({
    isExpanded: true,
  })
  expect(onExpand).toBeCalled()
  wrapper.setState({
    isExpanded: false,
  })
  expect(onCollapse).toBeCalled()
})
