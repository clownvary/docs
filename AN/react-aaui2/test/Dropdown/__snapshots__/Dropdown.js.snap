// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render right thing and don't change unexpected 1`] = `
<div
  aria-expanded={false}
  className="dropdown dropdown--flat dropdown--m"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  role="combobox"
  style={undefined}
>
  <button
    aria-label="Select one..."
    className="dropdown__button collapse"
    onClick={[Function]}
    onMouseDown={[Function]}
    type="button"
  >
    <span
      className="dropdown__button-text"
      role="textbox"
    >
      Select one...
    </span>
    <span
      className="icon-chevron-down"
    />
  </button>
  <div>
    <ul
      className="dropdown__menu"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      style={
        Object {
          "maxHeight": undefined,
        }
      }
      tabIndex={0}
    >
      <li
        aria-selected={undefined}
        className={undefined}
        onClick={[Function]}
        role="option"
        tabIndex={0}
      >
        <a>
          Canada
        </a>
      </li>
      <li
        aria-selected={undefined}
        className={undefined}
        onClick={[Function]}
        role="option"
        tabIndex={0}
      >
        <a>
          China
        </a>
      </li>
      <li
        aria-selected={undefined}
        className={undefined}
        onClick={[Function]}
        role="option"
        tabIndex={0}
      >
        <a>
          Japan
        </a>
      </li>
      <li
        aria-selected={undefined}
        className={undefined}
        onClick={[Function]}
        role="option"
        tabIndex={0}
      >
        <a>
          United States of America
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`AppWithFilter should render right thing and don't change unexpected 1`] = `
<div
  aria-expanded={false}
  className="dropdown dropdown--with-search dropdown--flat dropdown--m dropdown--with-search"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  role="combobox"
  style={undefined}
>
  <button
    aria-label="Select one..."
    className="dropdown__button collapse"
    onClick={[Function]}
    onMouseDown={[Function]}
    type="button"
  >
    <span
      className="dropdown__button-text"
      role="textbox"
    >
      Select one...
    </span>
    <span
      className="icon-chevron-down"
    />
  </button>
  <div>
    <ul
      className="dropdown__menu"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      style={
        Object {
          "maxHeight": undefined,
        }
      }
      tabIndex={0}
    >
      <li>
        <div
          className="dropdown__menu__search-box"
        >
          <i
            className="icon-search"
          />
          <input
            aria-label="Search"
            autoComplete="off"
            className="input"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Filter..."
            role="textbox"
            type="text"
          />
        </div>
      </li>
      <li
        className="dropdown__menu-divider"
        role="separator"
      />
      <li
        aria-selected={undefined}
        className={undefined}
        onClick={[Function]}
        role="option"
        tabIndex={0}
      >
        <a>
          Canada
        </a>
      </li>
      <li
        aria-selected={undefined}
        className={undefined}
        onClick={[Function]}
        role="option"
        tabIndex={0}
      >
        <a>
          China
        </a>
      </li>
      <li
        aria-selected={undefined}
        className={undefined}
        onClick={[Function]}
        role="option"
        tabIndex={0}
      >
        <a>
          Japan
        </a>
      </li>
      <li
        aria-selected={undefined}
        className={undefined}
        onClick={[Function]}
        role="option"
        tabIndex={0}
      >
        <a>
          United States of America
        </a>
      </li>
    </ul>
  </div>
</div>
`;
